From aaec62813fc00715f5f7784984895554f59e215b Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:43:13 +0800
Subject: [PATCH 4/4] GSI Android 17 compatibility:
 0004-Revert-Wi-Fi-Check-if-domain-field-is-not-empty-when.patch

Change-Id: I5562dc760fcdf6bbd767d09d9541f7882e724346
---
 .../android/settings/wifi/WifiConfigController.java   | 11 ++++++-----
 .../android/settings/wifi/WifiConfigController2.java  | 11 ++++++-----
 2 files changed, 12 insertions(+), 10 deletions(-)

diff --git a/src/com/android/settings/wifi/WifiConfigController.java b/src/com/android/settings/wifi/WifiConfigController.java
index 9d10abc..e466f09 100644
--- a/src/com/android/settings/wifi/WifiConfigController.java
+++ b/src/com/android/settings/wifi/WifiConfigController.java
@@ -534,11 +534,11 @@ public class WifiConfigController implements TextWatcher,
                 // Disallow submit if the user has not selected a CA certificate for an EAP network
                 // configuration.
                 enabled = false;
-            } else if (!caCertSelection.equals(mDoNotValidateEapServerString)
+            } else if (caCertSelection.equals(mUseSystemCertsString)
                     && mEapDomainView != null
                     && mView.findViewById(R.id.l_domain).getVisibility() != View.GONE
                     && TextUtils.isEmpty(mEapDomainView.getText().toString())) {
-                // Disallow submit if the user chooses to use a certificate for EAP server
+                // Disallow submit if the user chooses to use a system certificate for EAP server
                 // validation, but does not provide a domain.
                 enabled = false;
             }
@@ -572,12 +572,13 @@ public class WifiConfigController implements TextWatcher,
                 // Display warning if user chooses not to validate the EAP server with a
                 // user-supplied CA certificate in an EAP network configuration.
                 mView.findViewById(R.id.no_ca_cert_warning).setVisibility(View.VISIBLE);
-            } else if (!caCertSelection.equals(mUnspecifiedCertString)
+            } else if (caCertSelection.equals(mUseSystemCertsString)
                     && mEapDomainView != null
                     && mView.findViewById(R.id.l_domain).getVisibility() != View.GONE
                     && TextUtils.isEmpty(mEapDomainView.getText().toString())) {
-                // Display warning if user chooses to use a certificate without restricting the
-                // server domain that these certificates can be used to validate.
+                // Display warning if user chooses to use pre-installed public CA certificates
+                // without restricting the server domain that these certificates can be used to
+                // validate.
                 mView.findViewById(R.id.no_domain_warning).setVisibility(View.VISIBLE);
             }
         }
diff --git a/src/com/android/settings/wifi/WifiConfigController2.java b/src/com/android/settings/wifi/WifiConfigController2.java
index 9bfa928..1ce6451 100644
--- a/src/com/android/settings/wifi/WifiConfigController2.java
+++ b/src/com/android/settings/wifi/WifiConfigController2.java
@@ -642,11 +642,11 @@ public class WifiConfigController2 implements TextWatcher,
                 // Disallow submit if the user has not selected a CA certificate for an EAP network
                 // configuration.
                 enabled = false;
-            } else if (!caCertSelection.equals(mDoNotValidateEapServerString)
+            } else if (caCertSelection.equals(mUseSystemCertsString)
                     && mEapDomainView != null
                     && mView.findViewById(R.id.l_domain).getVisibility() != View.GONE
                     && TextUtils.isEmpty(mEapDomainView.getText().toString())) {
-                // Disallow submit if the user chooses to use a certificate for EAP server
+                // Disallow submit if the user chooses to use a system certificate for EAP server
                 // validation, but does not provide a domain.
                 enabled = false;
             }
@@ -680,12 +680,13 @@ public class WifiConfigController2 implements TextWatcher,
                 // Display warning if user chooses not to validate the EAP server with a
                 // user-supplied CA certificate in an EAP network configuration.
                 mView.findViewById(R.id.no_ca_cert_warning).setVisibility(View.VISIBLE);
-            } else if (!caCertSelection.equals(mUnspecifiedCertString)
+            } else if (caCertSelection.equals(mUseSystemCertsString)
                     && mEapDomainView != null
                     && mView.findViewById(R.id.l_domain).getVisibility() != View.GONE
                     && TextUtils.isEmpty(mEapDomainView.getText().toString())) {
-                // Display warning if user chooses to use a certificate without restricting the
-                // server domain that these certificates can be used to validate.
+                // Display warning if user chooses to use pre-installed public CA certificates
+                // without restricting the server domain that these certificates can be used to
+                // validate.
                 mView.findViewById(R.id.no_domain_warning).setVisibility(View.VISIBLE);
             }
         }
